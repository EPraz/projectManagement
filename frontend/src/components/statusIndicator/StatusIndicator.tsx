import { styled } from "@mui/material/styles";

interface StatusIndicatorProps {
  status: string;
  color: string;
}

export const StatusIndicator = styled("span")<StatusIndicatorProps>(
  ({ color }) => ({
    width: "8px",
    height: "8px",
    borderRadius: "50%",
    display: "inline-block",
    marginRight: "8px",
    backgroundColor: color,
  })
);
