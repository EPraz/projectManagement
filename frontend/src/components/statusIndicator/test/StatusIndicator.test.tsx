// StatusIndicator.test.tsx
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { StatusIndicator } from "../StatusIndicator";

describe("StatusIndicator", () => {
  it("match snapshot", () => {
    const { asFragment } = render(
      <StatusIndicator status="active" color="red" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a span with the correct background color", () => {
    // Renderizamos el componente con un color espec√≠fico
    const { container } = render(
      <StatusIndicator status="active" color="red" />
    );
    const spanElement = container.firstChild;
    expect(spanElement).toBeInTheDocument();
    // Verifica que el estilo se haya aplicado correctamente.
    expect(spanElement).toHaveStyle("background-color: red");
  });
});
