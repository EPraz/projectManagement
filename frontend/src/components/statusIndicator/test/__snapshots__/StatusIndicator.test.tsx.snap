// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIndicator match snapshot 1`] = `
<DocumentFragment>
  <span
    class="css-1m1p7l8"
    color="red"
  />
</DocumentFragment>
`;
